### PR TITLE
feat: filter substantial events

### PR DESCRIPTION
**Pull request by banusha2**
## Context

When many scheduled/periodic build events (annotation screwdriver.cd/buildPeriodically) which are shown in an event list, substantial events are not shown at first view.
## Objective
Screenshot 2023-04-12 at 10 26 04 AM

## References

https://github.com/screwdriver-cd/screwdriver/issues/2849
## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/pipeline-events/component.js [hunk 1]
```diff
@@ -445,6 +446,22 @@ export default Component.extend(ModelReloaderMixin, {
       return this.pipelineEvents;
     }
   }),
+  filteredEvents: computed('pipelineEvents', 'prEvents', 'currentEventType', 'filterSubstantialEvents', {
+    get() {
```
Comment (adong) [inline app/components/pipeline-events/component.js:450]: This `filteredEvents` adds unnecessary complexity :

The logics can go to `events: computed() ... ` in LINE
https://github.com/screwdriver-cd/ui/blob/fix/show-substantial-build-first/app/components/pipeline-events/component.js#L441

Comment (banusha2) [reply app/components/pipeline-events/component.js:441]: Done

File: app/components/pipeline-events/component.js [hunk 2]
```diff
@@ -436,10 +437,17 @@ export default Component.extend(ModelReloaderMixin, {
       return newPrEvents;
     }
   }),
-  events: computed('pipelineEvents', 'prEvents', 'currentEventType', {
+  events: computed('pipelineEvents', 'prEvents', 'currentEventType', 'filterSubstantialEvents', {
```
Comment (jithine) [inline app/components/pipeline-events/component.js:440]: please just call it `filterSchedulerEvents`

File: app/components/pipeline-events/template.hbs [hunk 3]
```diff
@@ -109,6 +109,15 @@
 
       <div class="tab-content">
         <tab.pane @activeId={{this.activeTab}} @id="events" @title="Events">
+          <div class="col-12" >
+            Toggle to filter substantial events:
```
Comment (jithine) [inline app/components/pipeline-events/template.hbs:113]: `Show Scheduler Events`

**Review by jithine — CHANGES_REQUESTED** [review]

**Comment by jithine** [comment]
Also how does this handle pagination when substantial events in the list are periodic builds.

**Review by tkyi — CHANGES_REQUESTED** [review]
Interaction should be exactly like this existing filter
Screenshot 2023-04-08 at 7 10 27 PM

Toggling would remove scheduled events from view

**Review by tkyi — CHANGES_REQUESTED** [review]
Interaction should be exactly like this existing filter
Screenshot 2023-04-08 at 7 10 27 PM

Toggling would remove scheduled events from view

File: app/components/pipeline-options/template.hbs [hunk 4]
```diff
@@ -389,6 +389,30 @@
             </div>
           </div>
         </li>
+        <li>
+          <div class="row">
+            <div class="col-10">
+              <h4>
+                Show Scheduler Events
```
Comment (tkyi) [inline app/components/pipeline-options/template.hbs:396]:
```suggestion
                Filter Scheduler Events
```

**Review by tkyi — APPROVED** [review]

**Review by jithine — APPROVED** [review]

**Comment by jithine** [comment]
Also we need to work with design team on how these filters which hides data from event list can be moved/shown right next to the events itself. So that user looking at it can know that it’s a filtered view and some data could be potenitally hidden.

**Pull request merged**